app/screens/onboarding/index.tsx

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect} from 'react';
+import {View, ListRenderItemInfo, useWindowDimensions, SafeAreaView, ScrollView, StyleSheet, Platform, NativeSyntheticEvent, NativeScrollEvent} from 'react-native';
+import {Navigation} from 'react-native-navigation';
+import Animated, {useAnimatedRef, useAnimatedStyle, useDerivedValue, useSharedValue, withTiming} from 'react-native-reanimated';
+
+import {storeOnboardingViewedValue} from '@actions/app/global';
+import {Screens} from '@app/constants';
+import Background from '@screens/background';
+import {goToScreen, loginAnimationOptions} from '@screens/navigation';
+import {OnboardingItem} from '@typings/screens/onboarding';
+
+import FooterButtons from './footer_buttons';
+import Paginator from './paginator';
+import SlideItem from './slide';
+import useSlidesData from './slides_data';
+
+import type {LaunchProps} from '@typings/launch';
+
+interface OnboardingProps extends LaunchProps {
+    theme: Theme;
+}
+
+const styles = StyleSheet.create({
+    onBoardingContainer: {
+        flex: 1,
+        alignItems: 'center',
+        justifyContent: 'center',
+        verticalAlign: 'top',
+    },
+    scrollContainer: {
+        flex: 1,
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+});
+
+const AnimatedSafeArea = Animated.createAnimatedComponent(SafeAreaView);
+
+const Onboarding = ({
+    theme,
+}: OnboardingProps) => {
+    const {width} = useWindowDimensions();
+    const {slidesData} = useSlidesData();
+    const LAST_SLIDE_INDEX = slidesData.length - 1;
+    const dimensions = useWindowDimensions();
+    const slidesRef = useAnimatedRef<ScrollView>();
+
+    const scrollX = useSharedValue(0);
+
+    // used to smothly animate the whole onboarding screen during the appear event scenario (from server screen back to onboarding screen)
+    const translateX = useSharedValue(dimensions.width);
+
+    const currentIndex = useDerivedValue(() => Math.round(scrollX.value / width));
```

Comment:
What is weird about it?

Reply:
currentIndex is used only outside of the UI thread (in a useCallback in this component), so in theory there is no need for it to be a shared value. It is a shared value because we want it to be "in sync" with scrollX and we have use `useDerivedValue`.

Not sure how `useDerivedValue` works, so not sure if the value on JS side is updated everytime it is called. If it does, this should be working without any issue. If not, this may create problems.

On top of that, changes in scrollX does not cause re-renders. So (unless something else causes the re-render, I have not looked deep enough into that) `nextSlide` will use always the same reference to currentIndex. But what is changing the value on the JS thread for the currentIndex if `useDerivedValue` is not being called?

Maybe I am missing something on how the internals of reanimated works, but all this seems a bit weird :P

Reply:
usedDerivedValue updates as it follows changes in scrollX shared values